python/core/auto_generated/processing/qgsprocessingparameterfieldform.sip.in

```diff
@@ -0,0 +1,106 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/processing/qgsprocessingparameterfieldform.h                *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+class QgsProcessingParameterFieldForm : QgsProcessingParameterDefinition
+{
+%Docstring
+A parameter for creating or reusing existing layer fields.
+
+.. versionadded:: 3.16
+%End
+
+%TypeHeaderCode
+#include "qgsprocessingparameterfieldform.h"
+%End
+  public:
+    QgsProcessingParameterFieldForm( const QString &name, const QString &description = QString(), const QString &parentLayerParameterName = QString(), bool optional = false );
+%Docstring
+Constructor for QgsProcessingParameterFieldForm.
+%End
+
+    static QString typeName();
+%Docstring
+Returns the type name for the parameter class.
+%End
+
+    virtual QgsProcessingParameterDefinition *clone() const;
+
+    virtual QString type() const;
+
+    virtual bool checkValueIsAcceptable( const QVariant &input, QgsProcessingContext *context = 0 ) const;
+
+    virtual QString valueAsPythonString( const QVariant &value, QgsProcessingContext &context ) const;
+
+    virtual QString asPythonString( QgsProcessing::PythonOutputType outputType = QgsProcessing::PythonQgsProcessingAlgorithmSubclass ) const;
+
+    virtual QVariantMap toVariantMap() const;
+
+    virtual bool fromVariantMap( const QVariantMap &map );
+
+    virtual QStringList dependsOnOtherParameters() const;
+
+
+    QString parentLayerParameterName() const;
+%Docstring
+Returns the name of the parent layer parameter, or an empty string if this is not set.
+
+.. seealso:: :py:func:`setParentLayerParameterName`
+%End
+
+    void setParentLayerParameterName( const QString &name );
+%Docstring
+Sets the ``name`` of the parent layer parameter. Use an empty string if this is not required.
+
+.. seealso:: :py:func:`parentLayerParameterName`
+%End
+
+};
+
+
+class QgsProcessingParameterTypeFieldForm : QgsProcessingParameterType
```

Comment:
This shouldn't be public -- it belongs in qgsprocessingparametertypeimpl.h instead

Reply:
I got heavily inspired by the implementation of `QgsProcessingParameterTypeFieldMap`. I thought `///@cond PRIVATE ... ///@endcond` would do the trick?

Reply:
That works to hide it from the generated API documentation, but in this case it's a hidden internal detail and so should not be exposed to Python at all. You can do this by adding a 

    #define SIP_NO_FILE

somewhere in the associated .h header and then re-running sipify_all.sh